### ISSUE TITLE
Request: training dataset for model pruning

### ISSUE THREAD

**Issue by BadBastion** [nikopueringer/CorridorKey#94]
Damn cool what you did with CorridorKey. I wanted to take a crack at shrinking the inference footprint with a structured channel pruning / quantization / distillation pass.

For an _ideal lobotomy_ its best to have all or at least some of the original training data. This gets used to compare the loss with the original model during the optimization process. If you folks are up for it, even a limited release of the training assets would make it much easier to do large optimization passes. 

Thanks!
 - Daniel - Some ML/Performance Engineer Guy.

**Comment by Mattie** [comment]
I think the world would love an optimized tiny model, as it's possible you could get this down to a really succinct model size.

**Comment by BadBastion** [comment]
@Mattie I should clarify since I linked generic pruning papers. The main bottleneck here is not really parameter count. 

The current model is already fairly modest in weight size. The main memory consumption comes from runtime activations, the data flowing through the model. GPU has to hold several huge intermediate tensors, including a couple 2048x2048 resolution passes. 

This is closer to pruning the residual/block width with standard res nets. Compressing the data flowing through the model and reducing the amount we need at any given time.
